### PR TITLE
Mono release resources on shutdown

### PR DESCRIPTION
**Pull request by Dunati**
This patch aims to clean up some allocated memory when shutting down the runtime.

### CODE REVIEW

**Comment by tritao** [comment]
whitelist

File: eglib/src/gmisc-win32.c [hunk 1]
```diff
@@ -137,30 +143,40 @@ g_get_home_dir (void)
...
 	return tmp_dir;
 }
 
+void
+g_env_cleanup (void)
+{
+	g_free (user_name);
+	g_free (home_dir);
+	g_free (tmp_dir);
+	tmp_dir = NULL;
```
Comment (migueldeicaza) [inline eglib/src/gmisc-win32.c]: This code will not really work, as tmp_dir at least has one scenario where a constant is assigned (just above).

**Comment by steveisok** [comment]
@lambdageek do we want to take this?

File: eglib/src/gmisc-win32.c [hunk 2]
```diff
@@ -137,30 +143,40 @@ g_get_home_dir (void)
...
-						tmp_dir = "C:\\temp";
-				}
+	if (tmp_dir == NULL) {
+		tmp_dir = g_getenv ("TMPDIR");
+		if (tmp_dir == NULL) {
+			tmp_dir = g_getenv ("TMP");
+			if (tmp_dir == NULL) {
+				tmp_dir = g_getenv ("TEMP");
+				if (tmp_dir == NULL)
+					tmp_dir = "C:\\temp";
```
Comment (lewing) [inline eglib/src/gmisc-win32.c]: Like miguel mentions bellow if this case is hit the g_free will fail

**Comment by lewing** [comment]
Closing due to no response.  Please reopen if you intend to keep working on this.

**Pull request closed without merging**